src/posix/main.c

```diff
@@ -101,15 +101,13 @@ void __gcov_flush();
  */
 enum
 {
-#if OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
     OT_POSIX_OPT_BACKBONE_INTERFACE_NAME = 'B',
-#endif
-    OT_POSIX_OPT_DEBUG_LEVEL    = 'd',
-    OT_POSIX_OPT_DRY_RUN        = 'n',
-    OT_POSIX_OPT_HELP           = 'h',
-    OT_POSIX_OPT_INTERFACE_NAME = 'I',
-    OT_POSIX_OPT_TIME_SPEED     = 's',
-    OT_POSIX_OPT_VERBOSE        = 'v',
+    OT_POSIX_OPT_DEBUG_LEVEL             = 'd',
```

Comment:
Explain:
The change of this file is not mandatory for this PR. However, the original code is incorrect because `OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE` is never defined in main.c. This change allows us to test Backbone features using `ot-cli` program.

Reply:
I guess `posix/main.c` is actually a application code thus can not get the core configuration?
Any comment on this? @bukepo